src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+ *          called in Python is as follows:
+ *
+ * @code
+ *          sin_thetas= numpy.array([0.05, 0.1, 0.15, ... ])
+ *          polar_quad = openmoc.Quadrature()
+ *          polar_quad.setNumPolarAngles(len(sin_thetas))
+ *          polar_quad.setSinThetas(sin_thetas)
+ * @endcode
+ *
+ * @param sin_thetas the array of sines of each polar angle
+ * @param num_polar the number of polar angles
+ */
+void Quadrature::setThetas(double* thetas, int num_azim_times_polar) {
+
+  if (_num_polar/2 * _num_azim/4 != num_azim_times_polar)
+    log_printf(ERROR, "Unable to set %d thetas for Quadrature "
+               "with %d polar angles and %d azimuthal angles"
+               " in each octant",
+               num_azim_times_polar, _num_polar/2, _num_azim/4);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_thetas != NULL) {
+    for (int i=0; i < _num_azim/2; i++)
+      delete [] _thetas[i];
+    delete [] _thetas;
+  }
+
+  /* Initialize memory for arrays */
+  _thetas = new double*[_num_azim/2];
+  for (int i=0; i < _num_azim/2; i++)
+    _thetas[i] = new double[_num_polar];
+
+  /* Extract sin thetas from user input */
+  int ap=0;
+  for (int a=0; a < _num_azim/4; a++) {
+    for (int p=0; p < _num_polar/2; p++) {
+      if (thetas[ap] < 0. || thetas[ap] > M_PI_2)
+        log_printf(ERROR, "Unable to set theta to %f which is "
+                   "not in the range [0,PI/2]", thetas[ap]);
+
+      _thetas[a][p] = thetas[ap];
+      _thetas[_num_azim/2 - a - 1][p] = thetas[ap];
+      _thetas[a][_num_polar - p - 1] = M_PI - thetas[ap];
+      _thetas[_num_azim/2 - a - 1][_num_polar - p - 1] = M_PI - thetas[ap];
+      ap++;
+    }
+  }
+}
+
+
+/**
+ * @brief Set the Quadrature's array of polar weights
+ * @param weights The polar weights
+ * @param num_azim_times_polar the total number of angles (azimuthal x polar)
+ */
+void Quadrature::setPolarWeights(FP_PRECISION* weights,
+                                 int num_azim_times_polar) {
+
+  if (_num_polar/2 * _num_azim/4 != num_azim_times_polar)
+    log_printf(ERROR, "Unable to set %d polar weights for Quadrature "
+               "with %d polar angles and %d azimuthal angles"
+               " in each octant",
+               num_azim_times_polar, _num_polar/2, _num_azim/4);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_polar_weights != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _polar_weights[a];
+    delete [] _polar_weights;
+  }
+
+  /* Initialize memory for arrays */
+  _polar_weights = new FP_PRECISION*[_num_azim/2];
+  for (int i=0; i < _num_azim/2; i++)
+    _polar_weights[i] = new FP_PRECISION[_num_polar];
+
+  /* Extract polar weights from user input */
+  int ap=0;
+  for (int a=0; a < _num_azim/4; a++) {
+    for (int p=0; p < _num_polar/2; p++) {
+      if (weights[ap] < 0. || weights[ap] > M_PI_2)
+        log_printf(ERROR, "Unable to polar weight to %f which is "
```

Comment:
No. The ``Quadrature::setPolarValues(...)`` is a private method that does not face with the user.